pkg/portal/middleware/aad.go

```diff
@@ -141,7 +142,18 @@ func (a *aad) AAD(h http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		session, err := a.store.Get(r, SessionName)
 		if err != nil {
-			a.internalServerError(w, err)
+			cookieError := err.(securecookie.Error)
+			if cookieError != nil && cookieError.IsDecode() {
+				cookie := &http.Cookie{
+					Name:    SessionName,
+					Path:    "/",
+					Expires: time.Now().Add(-1),
```

Comment:
It would be a better to use time.Unix(0,0) here. If there is time drift between the client and the server this may not work.

Reply:
Good call, done.